frontend/src/components/pages/Settings.tsx

```diff
@@ -405,36 +530,77 @@ const Settings = (): React.ReactElement => {
       if (!isValidEmail(emailsToValidate[i])) return false;
     }
 
-    if (numberOfKids !== "" && !isNonNegativeInt(numberOfKids)) return false;
+    if (!isNonNegativeInt(numKids)) return false;
 
     return true;
   };
 
+  const handleSaveSettings = async (
+    requestorId: string,
+    requestUserInfo: UserInfo,
+  ) => {
+    try {
+      const response = await updateUserByID({
+        variables: {
+          requestorId,
+          id: requestorId,
+          userInfo: requestUserInfo,
+        },
+      });
+
+      const newInfo = response.data.updateUserByID?.user?.info;
+      if (newInfo) {
+        setAuthenticatedUser({ ...authenticatedUser, info: newInfo });
+        setLocalStorageObjProperty(AUTHENTICATED_USER_KEY, "info", newInfo);
```

Comment:
need to override local storage to persist new changes on page refresh/navigation. without this change, the initial state of userinfo will not include new changes because local storage only updates after the user logs in

Reply:
Maybe we should write this as a comment?